expr/builtins/builtins.go

```diff
@@ -211,8 +211,29 @@ func Exists(ctx expr.EvalContext, item interface{}) (value.BoolValue, bool) {
 			return value.BoolValueTrue, true
 		}
 		return value.BoolValueFalse, true
-	case value.StringValue, value.BoolValue, value.NumberValue, value.StringsValue,
-		value.TimeValue, value.IntValue, value.SliceValue, value.MapIntValue:
+	case value.StringValue:
+		if node.Nil() {
+			return value.BoolValueFalse, true
+		}
+		return value.BoolValueTrue, true
+	case value.BoolValue:
+		return value.BoolValueTrue, true
+	case value.NumberValue:
+		if node.Nil() {
+			return value.BoolValueFalse, true
+		}
+		return value.BoolValueTrue, true
+	case value.IntValue:
+		if node.Nil() {
+			return value.BoolValueFalse, true
+		}
+		return value.BoolValueTrue, true
+	case value.TimeValue:
+		if node.Nil() {
+			return value.BoolValueFalse, true
+		}
+		return value.BoolValueTrue, true
```

Comment:
That 2nd value is "did this evaluate correctly", the first value is answering the question of "does it exist".   I don't like this Exists function in the vm, because it is inherently awkward, as the 2nd flag (bool) for the vm should aready answer does it exist.   Have thought about removing the function.